### PR TITLE
refactor: prep for composite action

### PR DESCRIPTION
**Pull request by thermokarst**

### CODE REVIEW

File: bin/library.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  --data "run_id=$GITHUB_RUN_ID" \
+  --data "artifact_name=$ARTIFACT_NAME" \
+  --header "Content-Type: application/x-www-form-urlencoded" \
+  --request POST https://library.qiime2.org/api/v1/packages/integrate/
+)
+
+code=$(echo $resp | grep HTTP | awk '{print $2}' )
+if [[ $code -ne 200 ]]
+then
+    echo $resp
```
Comment (thermokarst) [inline bin/library.sh:30]: Please add an initial error message here, followed by the `echo $resp`

File: bin/testing.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    -c conda-forge \
+    -c bioconda \
+    -c defaults \
+    --override-channels \
+    --strict-channel-priority \
+    --force-reinstall \
+    $PACKAGE_NAME
+else
+    # TODO: log error msg
+    # echo "::error ::Something went wrong"
```
Comment (thermokarst) [inline bin/testing.sh:57]: Please add an error message here.

File: bin/all.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+export Q2_CHANNEL=${Q2_CHANNEL}
+export ENV_URL=${ENV_URL}
+
+# Update when changing any subsequent bin/ scripts
+# action steps
+bash bin/setup.sh
+bash bin/build.sh
+node --unhandled-rejections=strict bin/artifact-upload/script.js
+bash bin/testing.sh
+bash bin/library.sh
```
[the remark this answers is not in the document]
Comment (lizgehret) [reply bin/all.sh:70]: Agreed - this is definitely a better place for all of these groupings. Thanks @thermokarst!

File: bin/all.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        export PLATFORM="osx"
+        ;;
+
+    Linux)
+        export ARTIFACT_NAME=linux-64
+        export PLATFORM="linux"
+        ;;
+
+    *)
+        # TODO: log error msg
```
[the remark this answers is not in the document]
Comment (lizgehret) [reply bin/all.sh:35]: Adding "ERROR: Compatible operating system not found"

File: main/script.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+async function main() {
+    try {
+        await exec.exec('bash ./bin/all.sh');
+    } catch (error) {
+        core.setFailed(error.message);
+    }
+}
+
+main().catch(core.setFailed);
```
Comment (thermokarst) [inline main/script.js:12]: Can you add a newline to the end of this file?

File: bin/artifact-upload/script.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        const files = await filesGlobber.glob();
+
+        const artifactClient = artifact.create();
+        artifactClient.uploadArtifact(process.env.ARTIFACT_NAME, files, process.env.BUILD_DIR);
+    } catch (error) {
+        core.setFailed(error.message);
+    }
+}
+
+artifactUpload().catch(core.setFailed);
```
Comment (thermokarst) [inline bin/artifact-upload/script.js:19]: Can you add a newline to the end of this file?

**Comment by lizgehret** [comment]
@thermokarst all requested updates have been added - let me know if I missed anything!

**Comment by lizgehret** [comment]
@thermokarst all requested changes have been updated! Let me know if I've missed anything!

File: bin/all.sh [hunk 7]
```diff
@@ -68,7 +68,9 @@ echo "::group::build.sh"
 bash bin/build.sh
 echo "::endgroup::"
 
+echo "::group::script.js"
```
Comment (thermokarst) [inline bin/all.sh:71]: We might wind up with more than one JS script - how about a more descriptive name, like `artifact-upload`?

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -17,11 +17,10 @@ jobs:
...
-    - run: |
-        npm install
-        npm run all
     - uses: ./
       with:
         recipe-path: ./example
         package-name: q2-nothing-to-see-here
         additional-tests: pytest --pyargs q2_nothing_to_see_here
+        build-target: tested
+        library-token: ${{ secrets.LIBRARY_TOKEN }}
```
Comment (thermokarst) [inline .github/workflows/test.yml:26]:
```suggestion
```

File: bin/testing.sh [hunk 9]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ -z $ADDITIONAL_TESTS ]]
+then
+    echo "Skipping additional tests"
```
Comment (thermokarst) [inline bin/testing.sh:7]:
```suggestion
    echo "Skipping, no additional tests specified"
```

**Pull request merged**